src/index/gpu_raft/gpu_raft_ivf_pq_config.h

```diff
@@ -14,62 +14,43 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#ifndef GPU_RAFT_IVF_PQ_CONFIG_H
+#define GPU_RAFT_IVF_PQ_CONFIG_H
 
-#ifndef IVF_RAFT_CONFIG_H
-#define IVF_RAFT_CONFIG_H
-
+#include "common/raft/integration/raft_knowhere_config.hpp"
+#include "common/raft/proto/raft_index_kind.hpp"
 #include "index/ivf/ivf_config.h"
 #include "knowhere/config.h"
 
 namespace knowhere {
 
-class RaftIvfFlatConfig : public IvfFlatConfig {
- public:
+struct GpuRaftIvfPqConfig : public IvfPqConfig {
+    CFG_FLOAT refine_ratio;
     CFG_INT kmeans_n_iters;
     CFG_FLOAT kmeans_trainset_fraction;
-    CFG_BOOL adaptive_centers;
-    KNOHWERE_DECLARE_CONFIG(RaftIvfFlatConfig) {
-        KNOWHERE_CONFIG_DECLARE_FIELD(k)
-            .set_default(10)
-            .description("search for top k similar vector.")
-            .set_range(1, 256)
-            .for_search();
-        KNOWHERE_CONFIG_DECLARE_FIELD(kmeans_n_iters)
-            .description("iterations to search for kmeans centers")
-            .set_default(20)
-            .for_train();
-        KNOWHERE_CONFIG_DECLARE_FIELD(kmeans_trainset_fraction)
-            .description("fraction of data to use in kmeans building")
-            .set_default(0.5)
-            .for_train();
-        KNOWHERE_CONFIG_DECLARE_FIELD(adaptive_centers)
-            .description("update centroids with new data")
-            .set_default(false)
-            .for_train();
-    }
-};
 
-class RaftIvfPqConfig : public IvfPqConfig {
- public:
-    CFG_INT streams_per_device;
-    CFG_INT kmeans_n_iters;
-    CFG_FLOAT kmeans_trainset_fraction;
-    CFG_INT m;
     CFG_STRING codebook_kind;
     CFG_BOOL force_random_rotation;
     CFG_BOOL conservative_memory_allocation;
     CFG_STRING lut_dtype;
     CFG_STRING internal_distance_dtype;
     CFG_FLOAT preferred_shmem_carveout;
-    KNOHWERE_DECLARE_CONFIG(RaftIvfPqConfig) {
+
+    KNOHWERE_DECLARE_CONFIG(GpuRaftIvfPqConfig) {
+        KNOWHERE_CONFIG_DECLARE_FIELD(refine_ratio)
+            .set_default(1.0f)
+            .description("search refine_ratio * k results then refine")
+            .for_search();
         KNOWHERE_CONFIG_DECLARE_FIELD(k)
             .set_default(10)
             .description("search for top k similar vector.")
-            .set_range(1, 1024)
+            .set_range(1, 1024)  // Declared in base but limited to 1024
```

Comment:
`(1, 1024)` is invalid. IVFPQ supports unlimited number of k and this IS IMPORTANT. I did experiments with RAFT's IVFPQ for k=100000. Please check